core/src/main/scala/com/salesforce/op/stages/impl/classification/OpXGBoostClassifier.scala

```diff
@@ -0,0 +1,187 @@
+
+  def setSketchEps(value: Double): this.type = set(sketchEps, value)
+
+  def setScalePosWeight(value: Double): this.type = set(scalePosWeight, value)
+
+  def setSampleType(value: String): this.type = set(sampleType, value)
+
+  def setNormalizeType(value: String): this.type = set(normalizeType, value)
+
+  def setRateDrop(value: Double): this.type = set(rateDrop, value)
+
+  def setSkipDrop(value: Double): this.type = set(skipDrop, value)
+
+  def setLambdaBias(value: Double): this.type = set(lambdaBias, value)
+
+  // setters for learning params
+  def setObjective(value: String): this.type = set(objective, value)
+
+  def setBaseScore(value: Double): this.type = set(baseScore, value)
+
+  def setEvalMetric(value: String): this.type = set(evalMetric, value)
+
+  def setTrainTestRatio(value: Double): this.type = set(trainTestRatio, value)
+
+  def setNumEarlyStoppingRounds(value: Int): this.type = set(numEarlyStoppingRounds, value)
+
+  def setCustomObj(value: ObjectiveTrait): this.type = set(customObj, value)
+
+  def setCustomEval(value: EvalTrait): this.type = set(customEval, value)
+
+}
+
+
+/**
+ * Class that takes in a spark [[XGBoostClassificationModel]] and wraps it into an OP model which returns a
+ * Prediction feature
+ *
+ * @param sparkModel    model to wrap
+ * @param operationName unique name of the operation this stage performs
+ * @param uid           uid to give stage
+ */
+class OpXGBoostClassificationModel
+(
+  sparkModel: XGBoostClassificationModel,
+  uid: String = UID[OpXGBoostClassificationModel],
+  operationName: String = classOf[XGBoostClassifier].getSimpleName
+)(
+  implicit tti1: TypeTag[RealNN],
+  tti2: TypeTag[OPVector],
+  tto: TypeTag[Prediction],
+  ttov: TypeTag[Prediction#Value]
+) extends OpProbabilisticClassifierModel[XGBoostClassificationModel](
+  sparkModel = sparkModel, uid = uid, operationName = operationName
+) {
+  import OpXGBoost._
+
+  protected def predictRawMirror: MethodMirror = throw new NotImplementedError()
+  protected def raw2probabilityMirror: MethodMirror = throw new NotImplementedError()
+  @transient lazy val probability2predictionMirror =
+    reflectMethod(getSparkMlStage().get, "probability2prediction")
+
+  private lazy val model = getSparkMlStage().get
+  private lazy val booster = model.nativeBooster
+  private lazy val treeLimit = model.getTreeLimit.toInt
+
+  override def transformFn: (RealNN, OPVector) => Prediction = (label, features) => {
+    val data = removeMissingValues(Iterator(features.value.asXGB), model.getMissing)
+    val dm = new DMatrix(dataIter = data)
+    // TODO: can we avoid two booster.predict calls here?
+    val rawPred = booster.predict(dm, outPutMargin = true, treeLimit = treeLimit)(0).map(_.toDouble)
+    val prob = booster.predict(dm, outPutMargin = false, treeLimit = treeLimit)(0).map(_.toDouble)
+    val probability = if (model.numClasses == 2) Array(1.0 - prob(0), prob(0)) else prob
```

Comment:
does XGboost support multiclass?

Reply:
Yes, Its the same class internally and is controlled by the num_class argument - https://github.com/dmlc/xgboost/blob/31d1baba3dbef251d2b5f4ae09ced8b3d29c8d13/jvm-packages/xgboost4j-example/src/main/scala/ml/dmlc/xgboost4j/scala/example/spark/SparkMLlibPipeline.scala

Should I add it?

Reply:
Yes we should add it for multiclass as well